commands/run/run.go

```diff
@@ -51,8 +54,27 @@ func runAccordingE2E() error {
 	logger.Log.Infof("setup part finished successfully")
 
 	// trigger part
+	err = trigger.DoActionAccordingE2E()
+	if err != nil {
+		return err
+	}
+	logger.Log.Infof("trigger part finished successfully")
 
 	// verify part
+	err = verify.DoVerifyAccordingConfig()
+	if err != nil {
+		return err
+	}
+	logger.Log.Infof("verify part finished successfully")
+
+	// cleanup part
+	defer func() {
+		err = cleanup.DoCleanupAccordingE2E()
+		if err != nil {
+			logger.Log.Errorf("cleanup part error: %s", err)
+		}
+		logger.Log.Infof("cleanup part finished successfully")
+	}()
```

Comment:
Should move this to right after the `setup` part.

Also, please consider adding an option (called `on`) to `cleanup` section to specify when we should clean up the resources, for example, `always`, `on-success`, so that users can see what happened when it failed

Reply:
Ok, i think its necessary, and i will add it in the next PR.